src/crimson/os/seastore/btree/fixed_kv_node.h

```diff
@@ -130,7 +130,7 @@ struct FixedKVNode : ChildableCachedExtent {
       children[offset] = child;
       set_child_ptracker(child);
     } else {
-      // this can only happen when reserving lba spaces
+      // this can happen when reserving lba spaces and cloning mappings
       ceph_assert(is_leaf_and_has_children());
       // this is to avoid mistakenly copying pointers from
       // copy sources when committing this lba node, because
```

Comment:
I can barely remember why the not-null reserved-ptr is explicitly needed and necessary.

Looking at `copy_children_from_stable_source()`, the major difference is that nullptr won't be copied from the stable source(s), and the reserved-ptr will be, still seems reasonable to me in supporting reserve-zero.

But I'm not following how this is useful for the cloning scenario.

Reply:
`reserved_ptr` is not specific to a certain usage. Its existence is to avoid wrongly copying out-dated child pointer from copy_sources. For example: if ObjectDataHandler removed a normal lba mapping and replaced it with a zero mapping or an indirect mapping, we shouldn't copy that normal mapping's child pointer to the new LBALeafNode. However, since both mappings have the same laddr, we don't know whether a change happened to the mapping in LBALeafNode. So, to identify the change, we insert a `reserved_ptr` for this kind of mapping.

Reply:
Thanks! I see `reserved_ptr` is to mark that this child has been updated (regardless whether the child needs to be an extent), so that upon commit, the according entries from copy_sources shouldn't overwrite the child because they are out-of-date.

Just to double confirm I'm fully understanding this. It still looks to me that there is no difference between a nullptr and reserved-ptr child for indirect's leaf nodes. Because they simply don't need to track child pointers, is that correct?